mbbsd/register.c

```diff
@@ -916,27 +935,24 @@ new_register(void)
 	}
     }
 
-    try = 0;
+    // Over 18.
     y++;
-    do {
-        char ans[3];
-	if (++try > 10) {
-	    vmsg(MSG_ERR_MAXTRIES);
-	    exit(1);
-	}
-        mvouts(y, 0, "��������ݪO�i�঳����Ť��e�u�A�X���~�H�h�\\Ū�C");
-        getdata(y + 1, 0,
-		"�z�O�_�~���Q�K���æP�N�[�ݦ����ݪO(�Y�_�п�Jn)? [y/n]:",
-                ans, 3, LCECHO);
-        if (ans[0] == 'y') {
-            newuser.over_18  = 1;
-            break;
-        } else if (ans[0] == 'n') {
-            newuser.over_18  = 0;
-            break;
-        }
-        bell();
-    } while (1);
+    mvouts(y, 0, "��������ݪO�i�঳����Ť��e�u�A�X���~�H�h�\\Ū�C");
+    if (query_yn(y + 1,
+		"�z�O�_�~���Q�K���æP�N�[�ݦ����ݪO(�Y�_�п�Jn)? [y/n]:"))
+	newuser.over_18 = 1;
+
+    // Whether to limit login to secure connection only.
+    if (mbbsd_is_secure_connection()) {
+	// Screen full.
+	y = 17;
+	move(y, 0); clrtobot();
+	outs("[ �s�u�]�w ]");
+
+	y++;
+	if (query_yn(y, "�z�O�_�n����b���ȯ�ϥΦw���s�u�n�J? [y/n]:"))
+	    newuser.uflag |= UF_SECURE_LOGIN;
```

Comment:
I thought you wanted to make this the default? lol

Reply:
Yeah, but later I thought that user might not know what a secure connection even is... So I made them to be responsible for their own choice instead.